packages/wallet-sdk/src/sign/SignRequestHandler.ts

```diff
@@ -114,6 +112,14 @@ export class SignRequestHandler implements RequestHandler {
   }
 
   async onDisconnect() {
+    this._signer = undefined;
     await this.signerConfigurator.onDisconnect();
   }
+
+  private async initSignerIfNeeded(): Promise<Signer> {
```

Comment:
not sure this is valid ts, but can we do something like
```suggestion
  private get signer(): Promise<Signer> {
```
returning a promise?

Reply:
yea.. tried but we cant do this for getter in TS yet

Reply:
@fan-zhang-sv try something like this https://github.com/coinbase/coinbase-wallet-sdk/pull/1169/files/f051aadda9be52512163b2eeff80c37448116c50#r1561853445
as long as you don't use `await`, i don't think ts has any issues with having computed getter with promise as return type.

Reply:
yea it works,

but when we try to use this signer, we'd either:
1. `(await (await this.signer).request(request);)`
2. or do `const signer = await this.signer` beforehand, which is similar to what we have now

still pretty troublesome? plus the getter become very hard to read
```
  private get signer(): Promise<Signer> {
    if (this._signer) return Promise.resolve(this._signer);

    return this.signerConfigurator.initSignerFromUserSelection().then((signer) => {
      this._signer = signer;
      return signer;
    });
  }
```

Reply:
gotcha. not so cleaner after all.